### ISSUE TITLE
accuracy is always 0 during training and testing

### ISSUE THREAD

**Issue by LitterQ** [soloice/Chinese-Character-Recognition#8]
hello, I train this Chinese-Character-Recognition with the repo's dataset, but the accuracy of train and test is 0 when train loss is declining.  Do you have any idea how to correct this?

**Comment by soloice** [comment]
Could you please post your loss/precision curve and weight distribution histogram? You can achieve this by opening your browser and tensorboard.

**Comment by LitterQ** [comment]
![image](https://user-images.githubusercontent.com/11575234/32832306-fc7d84ea-ca35-11e7-885c-f15ee36d7d16.png)
![image](https://user-images.githubusercontent.com/11575234/32832314-0672f05c-ca36-11e7-8cc2-a71aaa1b552f.png)
![image](https://user-images.githubusercontent.com/11575234/32832334-142e6bcc-ca36-11e7-98e8-42c633ff7354.png)
Those are loss and accuracy of validation, u can see the accuracy is 0 always.
![image](https://user-images.githubusercontent.com/11575234/32832478-95e2055c-ca36-11e7-8726-e08aeddb58d8.png)
And this is the histogram for training....

**Comment by soloice** [comment]
There must be something wrong with BatchNorm...
What's your TensorFlow version?

**Comment by LitterQ** [comment]
1.4, I think. And the code of the repo u mention can work, I think your code are similar, but ...

**Comment by whuchris** [comment]
I've the same problem with you, but i still cannot solve it. My train accuracy is normal, but the accuracy of validation shown in the tensorboard is like a roller coaster and is very slow. I use gpu to run but i don't know why. Everything goes ok but the accuracy. Is anything wrong with my tensorflow itself? I use 1.9.0 for ubuntu.

**Comment by soloice** [comment]
Many people meet this issue. I ran the code with TF 0.12 or 1.0.0rc, so it might have some compatibility issue with TF 1.5 or later version.
I'll check this later and try to make it work with the newest TF versions. Hopefully in a week or two.

**Comment by whuchris** [comment]
OK！ Thank you for your message, i will try too.    :)

**Comment by soloice** [comment]
Thanks for your attention!

**Comment by whuchris** [comment]
I'm a student from China, my teacher ask us to draw lessons from your code, hopefully we can solve this problem. :)

**Comment by whuchris** [comment]
Oh my god I think we've made it. When feeding the data to the graph, we should change the 'graph['is_training'] = True' to regularize the data. Then everything goes OK. Thank you very much! :)

**Comment by soloice** [comment]
I don't really get it.
If you use it for inference purpose, you should always set `graph['is_training'] = False`

**Comment by whuchris** [comment]
I use it when training. Exactly at the line 184.

**Comment by soloice** [comment]
That's totally fine. But you said the validation/test accuracy is abnormal, so we need to dive into the inference phase instead of training.

**Comment by whuchris** [comment]
Sorry I missed your message since My e-mail didn't remind me. In fact after I change the parameter everything goes OK including the inference. I've got about 88% accuracy in my own data.
